### DISCUSSION TITLE
Consume WIPs on Manufacturing Order

### DISCUSSION THREAD

**Discussion by sgfisme** [frePPLe/frepple#415] [Q&A] [answered]
Hi,
There is my trial model that has 3 sub-operations to make the product. I tried to model the WIP using Manufacturing Order, but it didn't be consumed, and created a new MO to fulfill the order. How can I configure my WIP that can be consumed?

![螢幕擷取畫面 2021-10-25 235910](https://user-images.githubusercontent.com/44051141/138730056-4ed71960-98b7-40cd-9e58-0cfa17298837.png)
![螢幕擷取畫面 2021-10-25 235318](https://user-images.githubusercontent.com/44051141/138729151-b9ddf094-52eb-41d4-b944-478316113065.png)

**Comment by jdetaeye** [comment]
Seeing these duplicate arrows here, I think your routing operation isn't correct.  
Either you have too many operationmaterial records, either you have also set the item field on the suboperations...

![image](https://user-images.githubusercontent.com/5219207/138730878-65d94e59-510a-4761-a894-9d67e9190670.png)

**Reply by sgfisme** [reply]
Thanks for your reply @jdetaeye .
Based on your suggestion, I simply my model as below and put my WIP on OP_Engine_Sub1. But the plan didn't consume the WIP on OP_Engine_Sub1. How can I model my operations or manufacturing order if I hope the WIP could be consumed? Because I hope the plan can calculate the remaining cycle time through sub-operation. Or I split my operation routing into different fixed_time operations?

![螢幕擷取畫面 2021-10-26 230115](https://user-images.githubusercontent.com/44051141/138906954-cdc68a14-a3be-4ccb-868e-8f12c65d46f4.png)
![螢幕擷取畫面 2021-10-26 225728](https://user-images.githubusercontent.com/44051141/138907338-4d9d23b0-5140-4c53-b4b4-762517c3cb35.png)

Just wondering to know that the supply path like figure 1 as above when I put the source material on the main operation. Is it correct?

![螢幕擷取畫面 2021-10-26 231446](https://user-images.githubusercontent.com/44051141/138909055-fd48098c-64c0-48bd-a223-c182aefd7f2b.png)
 
![螢幕擷取畫面 2021-10-26 230222](https://user-images.githubusercontent.com/44051141/138909078-d1ae7c80-64dc-4e05-a0c5-4607f6366533.png)

**Comment by jdetaeye** [comment, marked as answer]
- putting the child manufacturing order as "completed" while the parent routing manufacturing order is still "proposed" is not good.  Normally the parent should be in the status "approved" or "confirmed".
- Your operation table is ok.
- In the operationmaterial table, use the sub operations to produce and consume the material.

**Reply by sgfisme** [reply]
Thanks for your answer @jdetaeye .

**Reply by sgfisme** [reply]
Hi, @jdetaeye ,

I tried to put the child manufacturing order as "completed", but the WIP's quantities are small than Sales Order's quantities. The parent routing set status as "approved". But the Sales Order didn't fulfill from WIPs and created a purchase order that the quantities are the same as the Sales Order, and original WIPs were excess WIPs.

How could I set up the manufacturing order that can use WIPs to fulfill Sales Order, and create a purchase order that quantities are shortage quantities?
